src/components/subscriptions/data/hooks.js

```diff
@@ -54,16 +54,16 @@ export const useSubscriptions = ({ enterpriseId, errors, setErrors }) => {
         setSubscriptions(subscriptionsData);
       } catch (err) {
         logError(err);
-        setErrors({
-          ...errors,
+        setErrors(s => ({
+          ...s,
           [SUBSCRIPTIONS]: NETWORK_ERROR_MESSAGE,
-        });
+        }));
       } finally {
         setLoading(false);
       }
     };
     fetchData();
-  }, [enterpriseId, errors, setErrors]);
+  }, [enterpriseId, setErrors]);
```

Comment:
This function relies on both reading `errors` but also mutating `errors` via `setErrors`. Due to this, `errors` needs to be in the dependency array which means the function reference is re-computed whenever the API throws an error. This has a cascading effect in that the `useEffect` below that calls `loadCustomerAgreementData` is constantly fired.